src/tests/unit/doughs-dao.test.js

```diff
@@ -179,4 +184,52 @@ describe('test doughs dao', () => {
       });
     });
   });
+  context('getDoughById', () => {
+    let result;
+    beforeEach(() => {
+      connectionStub = sinon.stub().returns(executeReturn);
+      doughsDao = proxyquireDoughsDao(
+        connectionStub,
+        serializeDoughsStub,
+        serializeDoughStub,
+      );
+    });
+    context('when called with an integer id', () => {
+      beforeEach(async () => {
+        result = doughsDao.getDoughById(1);
+        await result;
+      });
+
+      it('calls execute with the right query and bind params', () => {
+        connectionStub.should.have.been.calledWith(
+          getDoughByIdQuery,
+          { id: 1 },
+        );
+      });
+
+      it('extracts rows from the return and passes them to the serializer', () => {
+        serializeDoughStub.should.have.been.calledWith(
+          executeReturnRows,
+          'doughs/1',
+        );
+      });
+
+      it('returns the result from the serializer', async () => result
+        .should.eventually.deep.equal(baseGetDoughsReturn));
+
+      context('when the database returns an empty result', () => {
+        beforeEach(async () => {
+          connectionStub.returns(emptyDatabaseReturn);
+          await doughsDao.getDoughById(1);
+        });
+
+        it('passes an empty array to the serializer', () => {
+          serializeDoughStub.should.have.been.calledWith(
+            [],
+            'doughs/1',
```

Comment:
Think about it more (especially from BDD point of view), I don't think we should even pass the empty array to the serializer, instead, if we got an empty raw data from DAO, we should directly return 404 response. What do you think?

Reply:
I wrote this originally so that the logic for determining that empty result == 404 happens in the path code instead of the DAO code. I could change it to do something like throwing an exception, but in that case we'd have to do something like defining a custom exception so that the path code didn't return a 500 when the record wasn't found.

Reply:
That could work, or take a look at this [example](https://github.com/osu-mist/terms-api/blob/b9acb21bb42fa5a483af15f514ea9017f4366c9f/src/api/v1/db/oracledb/terms-dao.js#L50-L69).

Reply:
Sounds good, I think that's basically what I'm going to go for.